aslo4/generator.py

```diff
@@ -841,14 +841,26 @@ def generate_web_page(
 
             # update the index files
             logger.debug("[STATIC][{}] Adding JSON".format(bundle.get_name()))
-            self.index.append(
-                bundle.generate_fingerprint_json(unique_icons=args.unique_icons)
-            )
+            
+            # Check if this bundle_id already exists in the index to prevent duplicates
+            bundle_id = bundle.get_bundle_id()
+            bundle_exists = False
+            for idx, existing_bundle in enumerate(self.index):
+                if existing_bundle.get('bundle_id') == bundle_id:
+                    # Update the existing entry instead of adding a duplicate
+                    self.index[idx] = bundle.generate_fingerprint_json(unique_icons=args.unique_icons)
+                    bundle_exists = True
+                    break
```

Comment:
You can check if the bundle exists without looping through `self.index`, looping through just to check for the existence of a bundle id is inefficient.